testfiles/PythonPluginMultiSpeedHeatPumpOverrideControl.py

```diff
@@ -0,0 +1,121 @@
+        self.cooling_setpoint_handle = None
+        self.coil_speed_level_handle = None
+        self.coil_speed_override_report_handle = None
+
+    def on_inside_hvac_system_iteration_loop(self, state) -> int:
+
+        # api is ready to execute
+        if self.api.exchange.api_data_fully_ready(state):
+
+            # get variable handles if needed
+            if self.need_to_get_handles:
+                self.zone_air_temp_handle = self.api.exchange.get_variable_handle(state,
+                                                                                  "Zone Air Temperature",
+                                                                                  "LIVING ZONE")
+
+                self.heating_setpoint_handle = self.api.exchange.get_variable_handle(state,
+                                                                                     "Zone Thermostat Heating Setpoint Temperature",
+                                                                                     "LIVING ZONE")
+
+                self.cooling_setpoint_handle = self.api.exchange.get_variable_handle(state,
+                                                                                     "Zone Thermostat Cooling Setpoint Temperature",
+                                                                                     "LIVING ZONE")
+
+                self.coil_speed_level_handle = self.api.exchange.get_actuator_handle(state,
+                                                                                     "Coil Speed Control",
+                                                                                     "Unitary System DX Coil Speed Value",
+                                                                                     "TWOSPEED HEAT PUMP 1")
+
+
+                self.coil_speed_override_report_handle = self.api.exchange.get_global_handle(state, "CoilSpeedLevelOverrideReport")
+
+                self.need_to_get_handles = False
+
+            # calculate
+            zone_air_temp = self.api.exchange.get_variable_value(state, self.zone_air_temp_handle)
+            heating_setpoint = self.api.exchange.get_variable_value(state, self.heating_setpoint_handle)
+            cooling_setpoint = self.api.exchange.get_variable_value(state, self.cooling_setpoint_handle)
+            if zone_air_temp < heating_setpoint - 2:
+                self.api.exchange.set_actuator_value(state, self.coil_speed_level_handle, 2.0)
+                self.api.exchange.set_global_value(state, self.coil_speed_override_report_handle, 1.0)
+            elif zone_air_temp < heating_setpoint:
+                self.api.exchange.set_actuator_value(state, self.coil_speed_level_handle, 1.0)
+                self.api.exchange.set_global_value(state, self.coil_speed_override_report_handle, 1.0)
+            elif zone_air_temp > cooling_setpoint:
+                self.api.exchange.set_actuator_value(state, self.coil_speed_level_handle, 1.0)
```

Comment:
Is the intent is for the python version to be the same as the EMS version? The python controls are using different speed levels and never use cooling speed 2.

Reply:
Modified them to be consistent.